File: Codinsa2015/Codinsa2015/Server/Views/EntityBaseView.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Microsoft.Xna.Framework.Graphics;
using Microsoft.Xna.Framework;


namespace Codinsa2015.Views
{

	public class EntityBaseView
	{

		static Encoding BOMLESS_UTF8 = new UTF8Encoding(false);
	
		/// <summary>
		/// Retourne la résistance magique effective de cette entité.
		/// </summary>
		public float GetMagicResist;	
		/// <summary>
		/// Retourne la valeur d'AP effective de cette entité.
		/// </summary>
		public float GetAbilityPower;	
		/// <summary>
		/// Retourne la valeur de CDR effective de cette entité. (de 0 à 0.40)
		/// </summary>
		public float GetCooldownReduction;	
		/// <summary>
		/// Obtient la vitesse de déplacement de l'entité.
		/// </summary>
		public float GetMoveSpeed;	
		/// <summary>
		/// Obtient la vitesse d'attaque effective de l'entité.
		/// </summary>
		public float GetAttackSpeed;	
		/// <summary>
		/// Obtient la vitesse d'attaque effective de l'entité.
		/// </summary>
		public float GetHPRegen;	
		/// <summary>
		/// Obtient les points d'attaque effectifs de cette entité.
		/// </summary>
		public float GetAttackDamage;	
		/// <summary>
		/// Fonction utilisée pour obtenir les points d'armure effectifs sur cette unité.
		/// </summary>
		public float GetArmor;	
		/// <summary>
		/// Obtient les HP actuels de cette entité.
		/// </summary>
		public float GetHP;	
		/// <summary>
		/// Obtient les HP max actuels de cette entité.
		/// </summary>
		public float GetMaxHP;	
		/// <summary>
		/// niveau du passif unique
		/// </summary>
		public int UniquePassiveLevel;	
		/// <summary>
		/// passif unique de cette entité.
		/// </summary>
		public EntityUniquePassives UniquePassive;	
		/// <summary>
		/// Si cette entité est un héros, obtient le rôle de ce héros.
		/// </summary>
		public EntityHeroRole Role;	
		/// <summary>
		/// Obtient la liste des altérations d'état affectées à cette entité.
		/// </summary>
		public List<StateAlterationView> StateAlterations;	
		/// <summary>
		/// Représente les points d'armure de base de cette entité.
		/// </summary>
		public float BaseArmor;	
		/// <summary>
		/// Représente la direction de cette entité.
		/// </summary>
		public Vector2 Direction;	
		/// <summary>
		/// Position de l'entité sur la map.
		/// </summary>
		public Vector2 Position;	
		/// <summary>
		/// Points de bouclier de cette entité.
		/// </summary>
		public float ShieldPoints;	
		/// <summary>
		/// Obtient les points de vie actuels de l'entité
		/// </summary>
		public float HP;	
		/// <summary>
		/// régénération de HP / s de base de cette unité.
		/// </summary>
		public float BaseHPRegen;	
		/// <summary>
		/// Obtient le nombre de points de vie maximum de base de cette entité.
		/// </summary>
		public float BaseMaxHP;	
		/// <summary>
		/// Obtient la vitesse de déplacement de base de l'entité.
		/// </summary>
		public float BaseMoveSpeed;	
		/// <summary>
		/// Retourne une valeur indiquant si l'entité est morte.
		/// </summary>
		public bool IsDead;	
		/// <summary>
		/// Retourne le type de cette entité.
		/// </summary>
		public EntityType Type;	
		/// <summary>
		/// Obtient l'id de cette entité.
		/// </summary>
		public int ID;	
		/// <summary>
		/// Obtient ou définit les points d'attaque de base de cette unité.
		/// </summary>
		public float BaseAttackDamage;	
		/// <summary>
		/// Cooldown reduction de base de cette unité.
		/// </summary>
		public float BaseCooldownReduction;	
		/// <summary>
		/// Attack speed de base de cette entité.
		/// </summary>
		public float BaseAttackSpeed;	
		/// <summary>
		/// Points d'AP de base de cette entité.
		/// </summary>
		public float BaseAbilityPower;	
		/// <summary>
		/// Point de résistance magique de base de cette entité.
		/// </summary>
		public float BaseMagicResist;	
		/// <summary>
		/// Obtient une valeur indiquant si cette entité est Rootée. (ne peut plus bouger).
		/// </summary>
		public bool IsRooted;	
		/// <summary>
		/// Obtient une valeur indiquant si cette unité est Silenced (ne peut pas utiliser de sorts).
		/// </summary>
		public bool IsSilenced;	
		/// <summary>
		/// Obtient une valeur indiquant si cette unité est Stuned (ne peut pas bouger ni utiliser de
		/// sorts).
		/// </summary>
		public bool IsStuned;	
		/// <summary>
		/// Obtient une valeur indiquant si cette unité possède une immunité temporaire aux dégâts.
		/// </summary>
		public bool IsDamageImmune;	
		/// <summary>
		/// Obtient une valeur indiquant si cette unité possède une immunité temporaire aux contrôles.
		/// </summary>
		public bool IsControlImmune;	
		/// <summary>
		/// Obtient une valeur indiquant si cette unité est aveuglé (ne peut pas lancer d'attaque avec son
		/// arme).
		/// </summary>
		public bool IsBlind;	
		/// <summary>
		/// Obtient une valeur indiquant si cette unité est invisible.
		/// </summary>
		public bool IsStealthed;	
		/// <summary>
		/// Obtient une valeur indiquant si cette entité possède la vision pure.
		/// </summary>
		public bool HasTrueVision;	
		/// <summary>
		/// Obtient une valeur indiquant si cette unité peut voir les wards.
		/// </summary>
		public bool HasWardVision;	
		/// <summary>
		/// Retourne la range à laquelle cette entité donne la vision.
		/// </summary>
		public float VisionRange;	
		public static EntityBaseView Deserialize(System.IO.StreamReader input) {
			EntityBaseView _obj =  new EntityBaseView();
			// GetMagicResist
			float _obj_GetMagicResist = Single.Parse(input.ReadLine());
			_obj.GetMagicResist = (float)_obj_GetMagicResist;
			// GetAbilityPower
			float _obj_GetAbilityPower = Single.Parse(input.ReadLine());
			_obj.GetAbilityPower = (float)_obj_GetAbilityPower;
			// GetCooldownReduction
			float _obj_GetCooldownReduction = Single.Parse(input.ReadLine());
			_obj.GetCooldownReduction = (float)_obj_GetCooldownReduction;
			// GetMoveSpeed
			float _obj_GetMoveSpeed = Single.Parse(input.ReadLine());
			_obj.GetMoveSpeed = (float)_obj_GetMoveSpeed;
			// GetAttackSpeed
			float _obj_GetAttackSpeed = Single.Parse(input.ReadLine());
			_obj.GetAttackSpeed = (float)_obj_GetAttackSpeed;
			// GetHPRegen
			float _obj_GetHPRegen = Single.Parse(input.ReadLine());
			_obj.GetHPRegen = (float)_obj_GetHPRegen;
			// GetAttackDamage
			float _obj_GetAttackDamage = Single.Parse(input.ReadLine());
			_obj.GetAttackDamage = (float)_obj_GetAttackDamage;
			// GetArmor
			float _obj_GetArmor = Single.Parse(input.ReadLine());
			_obj.GetArmor = (float)_obj_GetArmor;
			// GetHP
			float _obj_GetHP = Single.Parse(input.ReadLine());
			_obj.GetHP = (float)_obj_GetHP;
			// GetMaxHP
			float _obj_GetMaxHP = Single.Parse(input.ReadLine());
			_obj.GetMaxHP = (float)_obj_GetMaxHP;
			// UniquePassiveLevel
			int _obj_UniquePassiveLevel = Int32.Parse(input.ReadLine());
			_obj.UniquePassiveLevel = (int)_obj_UniquePassiveLevel;
			// UniquePassive
			EntityUniquePassives _obj_UniquePassive = (EntityUniquePassives)Int32.Parse(input.ReadLine());
			_obj.UniquePassive = (EntityUniquePassives)_obj_UniquePassive;
			// Role
			EntityHeroRole _obj_Role = (EntityHeroRole)Int32.Parse(input.ReadLine());
			_obj.Role = (EntityHeroRole)_obj_Role;
			// StateAlterations
			List<StateAlterationView> _obj_StateAlterations = new List<StateAlterationView>();
			int _obj_StateAlterations_count = Int32.Parse(input.ReadLine());
			for(int _obj_StateAlterations_i = 0; _obj_StateAlterations_i < _obj_StateAlterations_count; _obj_StateAlterations_i++) {
				StateAlterationView _obj_StateAlterations_e = StateAlterationView.Deserialize(input);
				_obj_StateAlterations.Add((StateAlterationView)_obj_StateAlterations_e);
			}
			_obj.StateAlterations = (List<StateAlterationView>)_obj_StateAlterations;
			// BaseArmor
			float _obj_BaseArmor = Single.Parse(input.ReadLine());
			_obj.BaseArmor = (float)_obj_BaseArmor;
			// Direction
			Vector2 _obj_Direction = Vector2.Deserialize(input);
			_obj.Direction = (Vector2)_obj_Direction;
			// Position
			Vector2 _obj_Position = Vector2.Deserialize(input);
			_obj.Position = (Vector2)_obj_Position;
			// ShieldPoints
			float _obj_ShieldPoints = Single.Parse(input.ReadLine());
			_obj.ShieldPoints = (float)_obj_ShieldPoints;
			// HP
			float _obj_HP = Single.Parse(input.ReadLine());
			_obj.HP = (float)_obj_HP;
			// BaseHPRegen
			float _obj_BaseHPRegen = Single.Parse(input.ReadLine());
			_obj.BaseHPRegen = (float)_obj_BaseHPRegen;
			// BaseMaxHP
			float _obj_BaseMaxHP = Single.Parse(input.ReadLine());
			_obj.BaseMaxHP = (float)_obj_BaseMaxHP;
			// BaseMoveSpeed
			float _obj_BaseMoveSpeed = Single.Parse(input.ReadLine());
			_obj.BaseMoveSpeed = (float)_obj_BaseMoveSpeed;
			// IsDead
			bool _obj_IsDead = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsDead = (bool)_obj_IsDead;
			// Type
			EntityType _obj_Type = (EntityType)Int32.Parse(input.ReadLine());
			_obj.Type = (EntityType)_obj_Type;
			// ID
			int _obj_ID = Int32.Parse(input.ReadLine());
			_obj.ID = (int)_obj_ID;
			// BaseAttackDamage
			float _obj_BaseAttackDamage = Single.Parse(input.ReadLine());
			_obj.BaseAttackDamage = (float)_obj_BaseAttackDamage;
			// BaseCooldownReduction
			float _obj_BaseCooldownReduction = Single.Parse(input.ReadLine());
			_obj.BaseCooldownReduction = (float)_obj_BaseCooldownReduction;
			// BaseAttackSpeed
			float _obj_BaseAttackSpeed = Single.Parse(input.ReadLine());
			_obj.BaseAttackSpeed = (float)_obj_BaseAttackSpeed;
			// BaseAbilityPower
			float _obj_BaseAbilityPower = Single.Parse(input.ReadLine());
			_obj.BaseAbilityPower = (float)_obj_BaseAbilityPower;
			// BaseMagicResist
			float _obj_BaseMagicResist = Single.Parse(input.ReadLine());
			_obj.BaseMagicResist = (float)_obj_BaseMagicResist;
			// IsRooted
			bool _obj_IsRooted = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsRooted = (bool)_obj_IsRooted;
			// IsSilenced
			bool _obj_IsSilenced = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsSilenced = (bool)_obj_IsSilenced;
			// IsStuned
			bool _obj_IsStuned = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsStuned = (bool)_obj_IsStuned;
			// IsDamageImmune
			bool _obj_IsDamageImmune = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsDamageImmune = (bool)_obj_IsDamageImmune;
			// IsControlImmune
			bool _obj_IsControlImmune = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsControlImmune = (bool)_obj_IsControlImmune;
			// IsBlind
			bool _obj_IsBlind = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsBlind = (bool)_obj_IsBlind;
			// IsStealthed
			bool _obj_IsStealthed = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.IsStealthed = (bool)_obj_IsStealthed;
			// HasTrueVision
			bool _obj_HasTrueVision = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.HasTrueVision = (bool)_obj_HasTrueVision;
			// HasWardVision
			bool _obj_HasWardVision = Int32.Parse(input.ReadLine()) == 0 ? false : true;
			_obj.HasWardVision = (bool)_obj_HasWardVision;
			// VisionRange
			float _obj_VisionRange = Single.Parse(input.ReadLine());
			_obj.VisionRange = (float)_obj_VisionRange;
			return _obj;
		}

		public void Serialize(System.IO.StreamWriter output) {
			// GetMagicResist
			output.WriteLine(((float)this.GetMagicResist).ToString());
			// GetAbilityPower
			output.WriteLine(((float)this.GetAbilityPower).ToString());
			// GetCooldownReduction
			output.WriteLine(((float)this.GetCooldownReduction).ToString());
			// GetMoveSpeed
			output.WriteLine(((float)this.GetMoveSpeed).ToString());
			// GetAttackSpeed
			output.WriteLine(((float)this.GetAttackSpeed).ToString());
			// GetHPRegen
			output.WriteLine(((float)this.GetHPRegen).ToString());
			// GetAttackDamage
			output.WriteLine(((float)this.GetAttackDamage).ToString());
			// GetArmor
			output.WriteLine(((float)this.GetArmor).ToString());
			// GetHP
			output.WriteLine(((float)this.GetHP).ToString());
			// GetMaxHP
			output.WriteLine(((float)this.GetMaxHP).ToString());
			// UniquePassiveLevel
			output.WriteLine(((int)this.UniquePassiveLevel).ToString());
			// UniquePassive
			output.WriteLine(((int)this.UniquePassive).ToString());
			// Role
			output.WriteLine(((int)this.Role).ToString());
			// StateAlterations
			output.WriteLine(this.StateAlterations.Count.ToString());
			for(int StateAlterations_it = 0; StateAlterations_it < this.StateAlterations.Count;StateAlterations_it++) {
				this.StateAlterations[StateAlterations_it].Serialize(output);
			}
			// BaseArmor
			output.WriteLine(((float)this.BaseArmor).ToString());
			// Direction
			this.Direction.Serialize(output);
			// Position
			this.Position.Serialize(output);
			// ShieldPoints
			output.WriteLine(((float)this.ShieldPoints).ToString());
			// HP
			output.WriteLine(((float)this.HP).ToString());
			// BaseHPRegen
			output.WriteLine(((float)this.BaseHPRegen).ToString());
			// BaseMaxHP
			output.WriteLine(((float)this.BaseMaxHP).ToString());
			// BaseMoveSpeed
			output.WriteLine(((float)this.BaseMoveSpeed).ToString());
			// IsDead
			output.WriteLine(this.IsDead ? 1 : 0);
			// Type
			output.WriteLine(((int)this.Type).ToString());
			// ID
			output.WriteLine(((int)this.ID).ToString());
			// BaseAttackDamage
			output.WriteLine(((float)this.BaseAttackDamage).ToString());
			// BaseCooldownReduction
			output.WriteLine(((float)this.BaseCooldownReduction).ToString());
			// BaseAttackSpeed
			output.WriteLine(((float)this.BaseAttackSpeed).ToString());
			// BaseAbilityPower
			output.WriteLine(((float)this.BaseAbilityPower).ToString());
			// BaseMagicResist
			output.WriteLine(((float)this.BaseMagicResist).ToString());
			// IsRooted
			output.WriteLine(this.IsRooted ? 1 : 0);
			// IsSilenced
			output.WriteLine(this.IsSilenced ? 1 : 0);
			// IsStuned
			output.WriteLine(this.IsStuned ? 1 : 0);
			// IsDamageImmune
			output.WriteLine(this.IsDamageImmune ? 1 : 0);
			// IsControlImmune
			output.WriteLine(this.IsControlImmune ? 1 : 0);
			// IsBlind
			output.WriteLine(this.IsBlind ? 1 : 0);
			// IsStealthed
			output.WriteLine(this.IsStealthed ? 1 : 0);
			// HasTrueVision
			output.WriteLine(this.HasTrueVision ? 1 : 0);
			// HasWardVision
			output.WriteLine(this.HasWardVision ? 1 : 0);
			// VisionRange
			output.WriteLine(((float)this.VisionRange).ToString());
		}

	}
}

File: Codinsa2015.Client/Codinsa2015.Client/Views/Clank/ClientCpp/src/EntityBaseView.cpp
#include "../inc/EntityBaseView.h"
void EntityBaseView::serialize(std::ostream& output) {
	// GetMagicResist
	output << ((float)this->GetMagicResist) << '\n';
	// GetAbilityPower
	output << ((float)this->GetAbilityPower) << '\n';
	// GetCooldownReduction
	output << ((float)this->GetCooldownReduction) << '\n';
	// GetMoveSpeed
	output << ((float)this->GetMoveSpeed) << '\n';
	// GetAttackSpeed
	output << ((float)this->GetAttackSpeed) << '\n';
	// GetHPRegen
	output << ((float)this->GetHPRegen) << '\n';
	// GetAttackDamage
	output << ((float)this->GetAttackDamage) << '\n';
	// GetArmor
	output << ((float)this->GetArmor) << '\n';
	// GetHP
	output << ((float)this->GetHP) << '\n';
	// GetMaxHP
	output << ((float)this->GetMaxHP) << '\n';
	// UniquePassiveLevel
	output << ((int)this->UniquePassiveLevel) << '\n';
	// UniquePassive
	output << ((int)this->UniquePassive) << '\n';
	// Role
	output << ((int)this->Role) << '\n';
	// StateAlterations
	output << this->StateAlterations.size() << '\n';
	for(int StateAlterations_it = 0; StateAlterations_it < this->StateAlterations.size(); StateAlterations_it++) {
		this->StateAlterations[StateAlterations_it].serialize(output);
	}

	// BaseArmor
	output << ((float)this->BaseArmor) << '\n';
	// Direction
	this->Direction.serialize(output);
	// Position
	this->Position.serialize(output);
	// ShieldPoints
	output << ((float)this->ShieldPoints) << '\n';
	// HP
	output << ((float)this->HP) << '\n';
	// BaseHPRegen
	output << ((float)this->BaseHPRegen) << '\n';
	// BaseMaxHP
	output << ((float)this->BaseMaxHP) << '\n';
	// BaseMoveSpeed
	output << ((float)this->BaseMoveSpeed) << '\n';
	// IsDead
	output << (this->IsDead ? 1 : 0) << '\n';
	// Type
	output << ((int)this->Type) << '\n';
	// ID
	output << ((int)this->ID) << '\n';
	// BaseAttackDamage
	output << ((float)this->BaseAttackDamage) << '\n';
	// BaseCooldownReduction
	output << ((float)this->BaseCooldownReduction) << '\n';
	// BaseAttackSpeed
	output << ((float)this->BaseAttackSpeed) << '\n';
	// BaseAbilityPower
	output << ((float)this->BaseAbilityPower) << '\n';
	// BaseMagicResist
	output << ((float)this->BaseMagicResist) << '\n';
	// IsRooted
	output << (this->IsRooted ? 1 : 0) << '\n';
	// IsSilenced
	output << (this->IsSilenced ? 1 : 0) << '\n';
	// IsStuned
	output << (this->IsStuned ? 1 : 0) << '\n';
	// IsDamageImmune
	output << (this->IsDamageImmune ? 1 : 0) << '\n';
	// IsControlImmune
	output << (this->IsControlImmune ? 1 : 0) << '\n';
	// IsBlind
	output << (this->IsBlind ? 1 : 0) << '\n';
	// IsStealthed
	output << (this->IsStealthed ? 1 : 0) << '\n';
	// HasTrueVision
	output << (this->HasTrueVision ? 1 : 0) << '\n';
	// HasWardVision
	output << (this->HasWardVision ? 1 : 0) << '\n';
	// VisionRange
	output << ((float)this->VisionRange) << '\n';
}

EntityBaseView EntityBaseView::deserialize(std::istream& input) {
	EntityBaseView _obj = EntityBaseView();
	// GetMagicResist
	float _obj_GetMagicResist; input >> _obj_GetMagicResist; input.ignore(1000, '\n');
	_obj.GetMagicResist = (float)_obj_GetMagicResist;
	// GetAbilityPower
	float _obj_GetAbilityPower; input >> _obj_GetAbilityPower; input.ignore(1000, '\n');
	_obj.GetAbilityPower = (float)_obj_GetAbilityPower;
	// GetCooldownReduction
	float _obj_GetCooldownReduction; input >> _obj_GetCooldownReduction; input.ignore(1000, '\n');
	_obj.GetCooldownReduction = (float)_obj_GetCooldownReduction;
	// GetMoveSpeed
	float _obj_GetMoveSpeed; input >> _obj_GetMoveSpeed; input.ignore(1000, '\n');
	_obj.GetMoveSpeed = (float)_obj_GetMoveSpeed;
	// GetAttackSpeed
	float _obj_GetAttackSpeed; input >> _obj_GetAttackSpeed; input.ignore(1000, '\n');
	_obj.GetAttackSpeed = (float)_obj_GetAttackSpeed;
	// GetHPRegen
	float _obj_GetHPRegen; input >> _obj_GetHPRegen; input.ignore(1000, '\n');
	_obj.GetHPRegen = (float)_obj_GetHPRegen;
	// GetAttackDamage
	float _obj_GetAttackDamage; input >> _obj_GetAttackDamage; input.ignore(1000, '\n');
	_obj.GetAttackDamage = (float)_obj_GetAttackDamage;
	// GetArmor
	float _obj_GetArmor; input >> _obj_GetArmor; input.ignore(1000, '\n');
	_obj.GetArmor = (float)_obj_GetArmor;
	// GetHP
	float _obj_GetHP; input >> _obj_GetHP; input.ignore(1000, '\n');
	_obj.GetHP = (float)_obj_GetHP;
	// GetMaxHP
	float _obj_GetMaxHP; input >> _obj_GetMaxHP; input.ignore(1000, '\n');
	_obj.GetMaxHP = (float)_obj_GetMaxHP;
	// UniquePassiveLevel
	int _obj_UniquePassiveLevel; input >> _obj_UniquePassiveLevel; input.ignore(1000, '\n');
	_obj.UniquePassiveLevel = (int)_obj_UniquePassiveLevel;
	// UniquePassive
	int _obj_UniquePassive_asInt; input >> _obj_UniquePassive; input.ignore(1000, '\n');
	EntityUniquePassives _obj_UniquePassive = (EntityUniquePassives)_obj_UniquePassive_asInt;
	_obj.UniquePassive = (::EntityUniquePassives)_obj_UniquePassive;
	// Role
	int _obj_Role_asInt; input >> _obj_Role; input.ignore(1000, '\n');
	EntityHeroRole _obj_Role = (EntityHeroRole)_obj_Role_asInt;
	_obj.Role = (::EntityHeroRole)_obj_Role;
	// StateAlterations
	std::vector<StateAlterationView> _obj_StateAlterations = std::vector<StateAlterationView>();
	int _obj_StateAlterations_count; input >> _obj_StateAlterations_count; input.ignore(1000, '\n');
	for(int _obj_StateAlterations_i = 0; _obj_StateAlterations_i < _obj_StateAlterations_count; _obj_StateAlterations_i++) {
		StateAlterationView _obj_StateAlterations_e = StateAlterationView::deserialize(input);
		_obj_StateAlterations.push_back((StateAlterationView)_obj_StateAlterations_e);
	}

	_obj.StateAlterations = (::std::vector<StateAlterationView>)_obj_StateAlterations;
	// BaseArmor
	float _obj_BaseArmor; input >> _obj_BaseArmor; input.ignore(1000, '\n');
	_obj.BaseArmor = (float)_obj_BaseArmor;
	// Direction
	Vector2 _obj_Direction = Vector2::deserialize(input);
	_obj.Direction = (::Vector2)_obj_Direction;
	// Position
	Vector2 _obj_Position = Vector2::deserialize(input);
	_obj.Position = (::Vector2)_obj_Position;
	// ShieldPoints
	float _obj_ShieldPoints; input >> _obj_ShieldPoints; input.ignore(1000, '\n');
	_obj.ShieldPoints = (float)_obj_ShieldPoints;
	// HP
	float _obj_HP; input >> _obj_HP; input.ignore(1000, '\n');
	_obj.HP = (float)_obj_HP;
	// BaseHPRegen
	float _obj_BaseHPRegen; input >> _obj_BaseHPRegen; input.ignore(1000, '\n');
	_obj.BaseHPRegen = (float)_obj_BaseHPRegen;
	// BaseMaxHP
	float _obj_BaseMaxHP; input >> _obj_BaseMaxHP; input.ignore(1000, '\n');
	_obj.BaseMaxHP = (float)_obj_BaseMaxHP;
	// BaseMoveSpeed
	float _obj_BaseMoveSpeed; input >> _obj_BaseMoveSpeed; input.ignore(1000, '\n');
	_obj.BaseMoveSpeed = (float)_obj_BaseMoveSpeed;
	// IsDead
	bool _obj_IsDead; input >> _obj_IsDead; input.ignore(1000, '\n');
	_obj.IsDead = (bool)_obj_IsDead;
	// Type
	int _obj_Type_asInt; input >> _obj_Type; input.ignore(1000, '\n');
	EntityType _obj_Type = (EntityType)_obj_Type_asInt;
	_obj.Type = (::EntityType)_obj_Type;
	// ID
	int _obj_ID; input >> _obj_ID; input.ignore(1000, '\n');
	_obj.ID = (int)_obj_ID;
	// BaseAttackDamage
	float _obj_BaseAttackDamage; input >> _obj_BaseAttackDamage; input.ignore(1000, '\n');
	_obj.BaseAttackDamage = (float)_obj_BaseAttackDamage;
	// BaseCooldownReduction
	float _obj_BaseCooldownReduction; input >> _obj_BaseCooldownReduction; input.ignore(1000, '\n');
	_obj.BaseCooldownReduction = (float)_obj_BaseCooldownReduction;
	// BaseAttackSpeed
	float _obj_BaseAttackSpeed; input >> _obj_BaseAttackSpeed; input.ignore(1000, '\n');
	_obj.BaseAttackSpeed = (float)_obj_BaseAttackSpeed;
	// BaseAbilityPower
	float _obj_BaseAbilityPower; input >> _obj_BaseAbilityPower; input.ignore(1000, '\n');
	_obj.BaseAbilityPower = (float)_obj_BaseAbilityPower;
	// BaseMagicResist
	float _obj_BaseMagicResist; input >> _obj_BaseMagicResist; input.ignore(1000, '\n');
	_obj.BaseMagicResist = (float)_obj_BaseMagicResist;
	// IsRooted
	bool _obj_IsRooted; input >> _obj_IsRooted; input.ignore(1000, '\n');
	_obj.IsRooted = (bool)_obj_IsRooted;
	// IsSilenced
	bool _obj_IsSilenced; input >> _obj_IsSilenced; input.ignore(1000, '\n');
	_obj.IsSilenced = (bool)_obj_IsSilenced;
	// IsStuned
	bool _obj_IsStuned; input >> _obj_IsStuned; input.ignore(1000, '\n');
	_obj.IsStuned = (bool)_obj_IsStuned;
	// IsDamageImmune
	bool _obj_IsDamageImmune; input >> _obj_IsDamageImmune; input.ignore(1000, '\n');
	_obj.IsDamageImmune = (bool)_obj_IsDamageImmune;
	// IsControlImmune
	bool _obj_IsControlImmune; input >> _obj_IsControlImmune; input.ignore(1000, '\n');
	_obj.IsControlImmune = (bool)_obj_IsControlImmune;
	// IsBlind
	bool _obj_IsBlind; input >> _obj_IsBlind; input.ignore(1000, '\n');
	_obj.IsBlind = (bool)_obj_IsBlind;
	// IsStealthed
	bool _obj_IsStealthed; input >> _obj_IsStealthed; input.ignore(1000, '\n');
	_obj.IsStealthed = (bool)_obj_IsStealthed;
	// HasTrueVision
	bool _obj_HasTrueVision; input >> _obj_HasTrueVision; input.ignore(1000, '\n');
	_obj.HasTrueVision = (bool)_obj_HasTrueVision;
	// HasWardVision
	bool _obj_HasWardVision; input >> _obj_HasWardVision; input.ignore(1000, '\n');
	_obj.HasWardVision = (bool)_obj_HasWardVision;
	// VisionRange
	float _obj_VisionRange; input >> _obj_VisionRange; input.ignore(1000, '\n');
	_obj.VisionRange = (float)_obj_VisionRange;
	return _obj;
}


